mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -252,8 +254,10 @@ object KMeansModel extends MLReadable[KMeansModel] {
       }
       val model = new KMeansModel(metadata.uid, new MLlibKMeansModel(clusterCenters))
       DefaultParamsReader.getAndSetParams(model, metadata)
-      DefaultParamsReader.loadInitialModel[KMeansModel](path, sc)
-        .foreach(v => model.set(model.initialModel, v))
+      DefaultParamsReader.loadInitialModel[KMeansModel](path, sc) match {
+        case Success(v) => model.set(model.initialModel, v)
+        case Failure(e) => if (!e.isInstanceOf[InvalidInputException]) throw e
```

Comment:
This can be

````scala
DefaultParamsReader.loadInitialModel[KMeansModel](path, sc) match {
        case Success(v) => model.set(model.initialModel, v)
        case Failure(_: InvalidInputException) => // initialModel doesn't exist, do nothing
        case Failure(e) => throw e
      }
````